Fixes for Search Functionality and Flair Usability in Forum.

brandonhach opened the pull request:
# Overview
After fetching new changes, I had to fix and restore the functionality for `search_post` and maintain usability across the forum for flairs. This new patch will fix issues where querying posts associated with the state of the forum page will not work. Additionally, the forum and each subforum page will have its specific description.

## User Stories
- As a forum user, I am searching on the main forum page to conduct a broad search across all posts.
- As a forum user,  I will use the search feature to find related posts tailored to that particular subforum.
- As a forum user, I want to edit the flairs in my post. 
- As a forum user, I want to update my post with more or less flairs.

## Features 
- `search_post` issues with querying is fixed.
- Associated forums or sub-forums will now have their personalized description which is populated from a new table called `forum_description`.
## Changes
- Refactored JavaScript logic for dynamic flair.
- Added a small CSS animation for each flair.
- Documented and written new table `forum_description` in `schema.sql`.
- Added `forum_description` table to `models.py`
- Commented out the `test/test_app.py` test because of failure and interference with my pytest.
- Implement a utility method called `get_forum_description` to assist in querying required and associated forum descriptions.
- Routes associated with serving endpoints for the forum and subforums, or those with a render template, now incorporate the get_forum_description method. This enhancement fetches and passes description data from the database to the respective forum web pages.

## Known issues or Bugs:
None

# Snapshots
![image](https://github.com/weiraven/for-the-board/assets/58790036/052fe6a3-425e-40a3-a48e-84064d000b20)
![image](https://github.com/weiraven/for-the-board/assets/58790036/656da9e8-b0d0-4451-b3e5-53682f69fce7)
![image](https://github.com/weiraven/for-the-board/assets/58790036/4dadb378-3f7e-41aa-b859-929c9827e746)
![image](https://github.com/weiraven/for-the-board/assets/58790036/bd1b9656-9f7a-42b1-bd46-0eb1727b3e28)


### Let me know if there's anything I missed.
![main-qimg-a27597ae181e14e3e448e7116be1c210](https://github.com/weiraven/for-the-board/assets/58790036/e9d960b1-969a-4e65-9360-fe167b021ed1)

weiraven reviewed (changes requested):
- Please revisit search by title. Comment posts should be filtered out of the results since they don't make sense outside of the context of their parent post. You should be able to implement this easily if you look at the extra filter I added to the forum/subforum functions which only loads parent posts and not their associated comments separately.

Flair search looks good but the div formatting in edit_post.html got screwed up a little. If you pull the latest edits I've made on my branch, wei-ftb-prog, I've fixed. I also made some minor edits to the table names and schema so that they conform better to the conventions we've been using all along in the project, i.e., changed forum_description to ForumDescription etc.

brandonhach commented:
> * Please revisit search by title. Comment posts should be filtered out of the results since they don't make sense outside of the context of their parent post. You should be able to implement this easily if you look at the extra filter I added to the forum/subforum functions which only loads parent posts and not their associated comments separately.
> 
> 
> Flair search looks good but the div formatting in edit_post.html got screwed up a little. If you pull the latest edits I've made on my branch, wei-ftb-prog, I've fixed. I also made some minor edits to the table names and schema so that they conform better to the conventions we've been using all along in the project, i.e., changed forum_description to ForumDescription etc.

Alright, the changes have been made. Had to fix the subforum function to use the same logic as you pointed out. Thank you for reviewing my pr Raven!

weiraven reviewed (approved):
Verified that search was working as expected now. Still some minor typos and formatting issues but I can sweep those on my next push.
